mustache.mjs

```diff
@@ -479,14 +479,25 @@ Context.prototype.lookup = function lookup (name) {
  * avoid the need to parse the same template twice.
  */
 function Writer () {
-  this.cache = {};
+  this.templateCache = {
+    _cache: {},
+    set: function set (key, value) {
+      this._cache[key] = value;
+    },
+    get: function get (key) {
+      return this._cache[key];
+    },
+    clear: function clear () {
+      this._cache = {};
+    }
+  };
 }
 
 /**
  * Clears all cached templates in this writer.
  */
 Writer.prototype.clearCache = function clearCache () {
-  this.cache = {};
+  this.templateCache !== undefined && this.templateCache.clear();
```

Comment:
Let's do this undefined-check the same way for consistency, meaning here and on line 511. Up to you which approach you prefer 😃

Reply:
Done, I had a quick look at the eslint docs to see if there was a rule to enforce this. But the closest is `no-undef` which also stops `undefined` from being used to initialise objects etc. So I have raised a new rule proposal in the ESLint repo: https://github.com/eslint/eslint/issues/12754

Reply:
Done, I had a quick look at the eslint docs to see if there was a rule to enforce this. But the closest is `no-undef` which also stops `undefined` from being used to initialise objects etc. So I have raised a new rule proposal in the ESLint repo: https://github.com/eslint/eslint/issues/12754